### PR TITLE
add useRatioPresenter config

### PR DESCRIPTION
**Pull request by iBinh**
fix #53

### CODE REVIEW

**Comment by guoyingtao** [comment]
@iBinh The ratio values in RatioSelector should update with the original image width/height ratio.
* If original image's height > its width, we use 2:3/3:4/3:5 etc
* If original image's width > its height, we use 3:2/4:3/5:3 etc
* When the rotation button is clicked, ratio values need to be changed as well.

File: Sources/Mantis/Mantis.swift [hunk 1]
```diff
@@ -91,7 +91,7 @@ public struct CropToolbarConfig {
     public var optionButtonFontSizeForPad: CGFloat = 20
     public var cropToolbarHeightForVertialOrientation: CGFloat = 44
     public var cropToolbarWidthForHorizontalOrientation: CGFloat = 80
-    
+    public var useRatioPresenter = true
```
Comment (guoyingtao) [inline Sources/Mantis/Mantis.swift:94]: For me, I would like to use a enum type like 
```swift
public enum RatioCandidatesShowType {
    case presentRatioList
    case alwaysShowRatioList
}
```

Then add a `ratioCandidatesShowType: RatioCandidatesShowType ` to CropToolbarConfig with the intialValue  . presentRatioList

File: Sources/Mantis/CropViewController/RatioItemView.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        titleLabel.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+        titleLabel.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(tap))
+        addGestureRecognizer(gesture)
+        
+        layer.cornerRadius = 10
+        clipsToBounds = true
+        
+    }
+    @objc private func tap() {
```
Comment (guoyingtao) [inline Sources/Mantis/CropViewController/RatioItemView.swift:62]: We need to add blank lines between functions.

File: Sources/Mantis/CropViewController/RatioSelector.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        self.originalRatioH = originalRatioH
+        self.ratios = ratios
+        setupViews()
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupViews()
+    }
+    required init?(coder: NSCoder) {
```
Comment (guoyingtao) [inline Sources/Mantis/CropViewController/RatioSelector.swift:29]: We need to add blank lines between functions.

File: Sources/Mantis/CropViewController/RatioSelector.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        super.init(coder: coder)
+        setupViews()
+    }
+    private let scrollView: UIScrollView = {
```
Comment (guoyingtao) [inline Sources/Mantis/CropViewController/RatioSelector.swift:33]: We need to add blank lines between functions.

File: Sources/Mantis/CropViewController/RatioSelector.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    func reset() {
+        for ratioView in stackView.arrangedSubviews as! [RatioItemView] {
+            ratioView.selected = self.originalRatioH == ratioView.ratio.ratioH ? true : false
+        }
+    }
+    private func addRatioItems() {
```
Comment (guoyingtao) [inline Sources/Mantis/CropViewController/RatioSelector.swift:58]: We need to add blank lines between functions.

**Comment by iBinh** [comment]
Fixed

> @iBinh The ratio values in RatioSelector should update with the original image width/height ratio.
> 
> * If original image's height > its width, we use 2:3/3:4/3:5 etc
> * If original image's width > its height, we use 3:2/4:3/5:3 etc
> * When the rotation button is clicked, ratio values need to be changed as well.

**Comment by guoyingtao** [comment]
@iBinh Looks good! I think we are almost there.
Can you center RatioSelector horizontally when the device is iPad?
Also, if you want, you can also add an example to show this new feature in the example project and update README.

image
image

**Comment by iBinh** [comment]
fixed

**Review by guoyingtao — APPROVED** [review]
Looks good.
Thanks for the PR! I will release a new version for it later.

**Pull request merged**